File: backend/planta-smart-4.0/src/main/java/com/smart/appsa/model/enums/CorTampa.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

import jakarta.persistence.EnumeratedValue;

public enum CorTampa {
    PRETO(1),
    VERMELHO(2),
    AZUL(3);

    @EnumeratedValue
    int value;

    @JsonCreator
    public static CorTampa fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (CorTampa c : values()) {
            if (parsed.equals(c.value)) return c;
        }
        throw new IllegalArgumentException("CorTampa inválida: " + value);
    }

    private CorTampa(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/CorTampa.ts
export enum CorTampa {
  Preto = 'preto',
  Vermelho = 'vermelho',
  Azul = 'azul',
}

export const CorTampaToInt: Record<CorTampa, number> = {
  [CorTampa.Preto]: 1,
  [CorTampa.Vermelho]: 2,
  [CorTampa.Azul]: 3,
};

export const CorTampaStringToEnum: Record<string, CorTampa> = {
  PRETO: CorTampa.Preto,
  VERMELHO: CorTampa.Vermelho,
  AZUL: CorTampa.Azul,
};